packages/react/src/utils/objectUtils.ts

```diff
@@ -0,0 +1,10 @@
+export const objectValuesEqual = <T extends { [key: string]: unknown }>(
+  object1: T,
+  object2: T,
```

Comment:
Could we maybe use the [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) type since we expect these to be objects?